File: SurperPm-agent-web/frontend/src/lib/schemas/topic.ts
import { z } from "zod";

export const topicSchema = z.object({
  id: z.number(),
  workspace_id: z.string(),
  name: z.string(),
  description: z.string().nullable().optional(),
  goal_id: z.number().nullable().optional(),
  session_name: z.string().nullable().optional(),
  repo_url: z.string().nullable().optional(),
  pinned: z.boolean().default(false),
  archived: z.boolean().default(false),
  created_at: z.string().optional().default(""),
  updated_at: z.string().optional().default(""),
});

export type Topic = z.infer<typeof topicSchema>;

export const topicListSchema = z.array(topicSchema);

File: SurperPm-agent-web/backend/app/models/topic.py
"""Topic table model — Slack-style discussion channels."""

from datetime import UTC, datetime

from sqlmodel import Field, SQLModel


class Topic(SQLModel, table=True):
    __tablename__ = "topic"

    id: int | None = Field(default=None, primary_key=True)
    workspace_id: str = Field(foreign_key="workspace.id", nullable=False)
    name: str = Field(nullable=False)
    description: str | None = Field(default=None)
    goal_id: int | None = Field(default=None, foreign_key="goal.id")
    repo_url: str | None = Field(default=None)
    pinned: bool = Field(default=False)
    archived: bool = Field(default=False)
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
